dso-l2/src/main/java/com/tc/l2/state/ElectionManagerImpl.java

```diff
@@ -249,16 +252,16 @@ private synchronized void electionStarted(Enrollment e, State serverState) {
     logger.info("Election Started : " + e);
   }
 
-  private NodeID doElection(NodeID myNodeId, boolean isNew, WeightGeneratorFactory weightsFactory, State currentState)
+  private NodeID doElection(NodeID myNodeId, Topology topology, boolean isNew, WeightGeneratorFactory weightsFactory, State currentState)
       throws GroupException, InterruptedException {
 
     // Step 1: publish to cluster NodeID, weight and election start
     Enrollment e = EnrollmentFactory.createEnrollment(myNodeId, isNew, weightsFactory);
-    electionStarted(e, currentState);
+    electionStarted(e, currentState, topology);
```

Comment:
let's lock down the topology here by getting the servers now and passing that to election started.

Reply:
Yes, topology is already locked in `StateManagerImpl.java#runElection()` and passed here in `ElectionContext`